CustomTracks/Backgrounds/CustomBackground.cs

```diff
@@ -22,6 +23,65 @@ public CustomBackground(AssetBundle bundle, string songPath) : base(bundle)
     public override GameObject Load(BackgroundContext ctx)
     {
         var bg = Bundle.LoadAsset<GameObject>("assets/_background.prefab");
+
+        // MacOS Shader Handling
+        // May need to be expanded to other platforms eventually, but for now only check for invalid shaders on mac
+        if (Application.platform == RuntimePlatform.OSXPlayer)
+        {
+            // first add base game shaders, which should NOT be overwritten by shadercache shaders
+            Dictionary<string, Shader> shaderCache = Plugin.Instance.ShaderHelper.BaseGameShaderCache;
+
+            foreach (var cachedShader in Plugin.Instance.ShaderHelper.ShaderCache)
+            {
+                if (!shaderCache.ContainsKey(cachedShader.Key)) shaderCache.Add(cachedShader.Key, cachedShader.Value);
```

Comment:
So this'll mutate `BaseGameShaderCache`, right? Is that intended? If so the two dictionaries could just be merged at startup.